org.osgi.util.pushstream/src/org/osgi/util/pushstream/ThresholdPushbackPolicy.java

```diff
@@ -0,0 +1,102 @@
+/*******************************************************************************
+ * Copyright (c) Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0 
+ *******************************************************************************/
+package org.osgi.util.pushstream;
+
+import java.util.concurrent.BlockingQueue;
+
+/**
+ * Provides a configurable {@link PushbackPolicy} implementations. It provides
+ * zero back pressure until the buffer reaches a threshold level. Once the
+ * threshold is reached a linear policy is applied. The steepness and starting
+ * level of the back pressure can be configured separately. When the threshold
+ * is exceeded the the initial value is returned. After that it is calculated by
+ * factoring the excess elements and multiplying them with the configured step
+ * added to the initial value.
+ * <p/>
+ * The step size can be zero, which makes it return the initial value as a fixed
+ * back pressure after the threshold is exceeded.
+ * 
+ * @param <T> The type of objects in the {@link PushEvent}
+ * @param <U> The type of the Queue used in the user specified buffer
+ */
+public class ThresholdPushbackPolicy<T, U extends BlockingQueue<PushEvent< ? extends T>>>
+		implements PushbackPolicy<T,U> {
+
+	private final int	threshold;
+	private final long	initial;
+	private final long	step;
+
+	/**
+	 * A simple configuration with an initial back pressure of one and and
+	 * increase step size of one
+	 * 
+	 * @param threshold the queue size limit after which back pressure will be
+	 *            applied
+	 * @throws IllegalArgumentException if any of the given values is lower then
+	 *             zero
+	 */
+	public ThresholdPushbackPolicy(int threshold) {
+		this(threshold, 1, 1);
```

Comment:
```suggestion
		this(threshold, 1L, 1L);
```

Proper long constants.

Reply:
done